SecureStore.setItemAsync causing error on Android API level 22

## 🐛 Bug Report

### Environment

<!-- In your terminal run `expo diagnostics` and paste the output here. -->
  Expo CLI 2.21.2 environment info:
    System:
      OS: macOS 10.14.5
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.10.0 - /usr/local/bin/node
      npm: 6.9.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.2 AI-181.5540.7.32.5056338
      Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^33.0.0 => 33.0.7 
      react: 16.8.3 => 16.8.3 
      react-native: https://github.com/expo/react-native/archive/sdk-33.0.0.tar.gz => 0.59.8 
    npmGlobalPackages:
      expo-cli: 2.21.2

Targeting a standalone Android build.
<!-- Please also let us know about your app's target (iOS, Android, Client, Standalone, ExpoKit) -->

### Steps to Reproduce
1. `expo init Test22` (and go through the prompts to create a simple managed app) OR see [this snack](https://snack.expo.io/@amcauliffebabelstreet/test22) OR clone [this repo](https://github.com/sealedHuman/test22)
1. `cd Test22`
1. `expo ba` (and go through the prompts)
1. Sideload the resulting apk to a Moto X running Android 5.1 via `adb install test22.apk`
1. Tap the button on the app's screen
<!--
  How would you describe your issue to someone who doesn’t know you or your project?
  Try to write a sequence of steps that anybody can repeat to see the issue.
  Be specific! If the bug cannot be reproduced, your issue may be closed.
-->

### Expected Behavior
The text field should change from "waiting..." to "you did it".
<!--
  How did you expect your project to behave?
  It’s fine if you’re not sure your understanding is correct.
  Just write down what you thought would happen.
-->

### Actual Behavior
Nothing occurs on the app, but logcat produces this error:
`E/ExpoSecureStore: Caught unexpected exception when writing to SecureStore
    java.lang.IllegalStateException: Can't generate certificate
        at android.security.AndroidKeyPairGenerator.generateKeyPair(AndroidKeyPairGenerator.java:136)
        at java.security.KeyPairGenerator$KeyPairGeneratorImpl.generateKeyPair(KeyPairGenerator.java:276)
        at expo.modules.securestore.SecureStoreModule$HybridAESEncrypter.initializeKeyStoreEntry(SecureStoreModule.java:499)
        at expo.modules.securestore.SecureStoreModule$HybridAESEncrypter.initializeKeyStoreEntry(SecureStoreModule.java:452)
        at expo.modules.securestore.SecureStoreModule.getKeyEntry(SecureStoreModule.java:314)
        at expo.modules.securestore.SecureStoreModule.setItemImpl(SecureStoreModule.java:112)
        at expo.modules.securestore.SecureStoreModule.setValueWithKeyAsync(SecureStoreModule.java:74)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at org.unimodules.a.c.invokeExportedMethod(ExportedModule.java:96)
        at org.unimodules.adapters.react.NativeModulesProxy.callMethod(NativeModulesProxy.java:117)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
        at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:158)
        at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
        at android.os.Handler.handleCallback(Handler.java:739)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:29)
        at android.os.Looper.loop(Looper.java:135)
        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:232)
        at java.lang.Thread.run(Thread.java:818)
     Caused by: java.lang.UnsupportedOperationException: private exponent cannot be extracted
        at com.android.org.conscrypt.OpenSSLRSAPrivateKey.getPrivateExponent(OpenSSLRSAPrivateKey.java:161)
        at org.spongycastle.jcajce.provider.asymmetric.rsa.RSAUtil.a(RSAUtil.java:63)
        at org.spongycastle.jcajce.provider.asymmetric.rsa.DigestSignatureSpi.engineInitSign(DigestSignatureSpi.java:95)
        at java.security.Signature$SignatureImpl.engineInitSign(Signature.java:679)
        at java.security.Signature.initSign(Signature.java:330)
        at com.android.org.bouncycastle.x509.X509Util.calculateSignature(X509Util.java:243)
        at com.android.org.bouncycastle.x509.X509V3CertificateGenerator.generate(X509V3CertificateGenerator.java:434)
        at com.android.org.bouncycastle.x509.X509V3CertificateGenerator.generate(X509V3CertificateGenerator.java:412)
        at android.security.AndroidKeyPairGenerator.generateKeyPair(AndroidKeyPairGenerator.java:133)`
<!--
  Did something go wrong?
  Is something broken, or not behaving as you expected?
  Describe this section in detail, and attach screenshots if possible.
  Don't just say "it doesn't work"!
-->

### Reproducible Demo
Note: you must build a standalone Android binary and run it on a device running API level < 23 in order to reproduce this.
* https://snack.expo.io/@amcauliffebabelstreet/test22
* https://github.com/sealedHuman/test22
<!--
  Please share a project that reproduces the issue.
  There are two ways to do it:

    * Create a new app using https://snack.expo.io/ and try to reproduce the issue in it.
      This is useful if you roughly know where the problem is, or can’t share the real code.

    * Or, copy your app and remove things until you’re left with the minimal reproducible demo.
      This is useful for finding the root cause. You may then optionally create a Snack.

  This is a good guide to creating bug demos: https://stackoverflow.com/help/mcve
  Once you’re done, copy and paste the link to the Snack or a public GitHub repository below:
-->

<!--
  What happens if you skip this step?

  Someone will read your bug report, and maybe will be able to help you,
  but it’s unlikely that it will get much attention from the team. Eventually,
  the issue will likely get closed in favor of issues that have reproducible demos.

  Please remember that:

    * Issues without reproducible demos have a very low priority.
    * The person fixing the bug would have to do that anyway. Please be respectful of their time.
    * You might figure out the issues yourself as you work on extracting it.

  Thanks for helping us help you!
-->

--- a/android/expoview/src/main/java/host/exp/exponent/experience/DetachedModuleRegistryAdapter.java
+++ b/android/expoview/src/main/java/host/exp/exponent/experience/DetachedModuleRegistryAdapter.java
@@ -4,14 +4,13 @@ import com.facebook.react.bridge.NativeModule;
 import com.facebook.react.bridge.ReactApplicationContext;
 
 import org.json.JSONObject;
+import org.unimodules.adapters.react.ReactModuleRegistryProvider;
+import org.unimodules.core.ModuleRegistry;
+import org.unimodules.core.interfaces.InternalModule;
+import org.unimodules.core.interfaces.RegistryLifecycleListener;
 
 import java.util.List;
 import java.util.Map;
-
-import org.unimodules.adapters.react.ReactModuleRegistryProvider;
-import org.unimodules.core.ModuleRegistry;
-import org.unimodules.core.interfaces.RegistryLifecycleListener;
-import org.unimodules.core.interfaces.InternalModule;
 
 import host.exp.exponent.ExponentManifest;
 import host.exp.exponent.kernel.ExperienceId;
@@ -20,6 +19,7 @@ import versioned.host.exp.exponent.modules.universal.ConstantsBinding;
 import versioned.host.exp.exponent.modules.universal.ExpoModuleRegistryAdapter;
 import versioned.host.exp.exponent.modules.universal.ScopedFileSystemModule;
 import versioned.host.exp.exponent.modules.universal.ScopedUIManagerModuleWrapper;
+import versioned.host.exp.exponent.modules.universal.SecureStoreModuleBinding;
 
 public class DetachedModuleRegistryAdapter extends ExpoModuleRegistryAdapter {
   public DetachedModuleRegistryAdapter(ReactModuleRegistryProvider moduleRegistryProvider) {
@@ -46,6 +46,15 @@ public class DetachedModuleRegistryAdapter extends ExpoModuleRegistryAdapter {
 
     // Overriding expo-file-system FileSystemModule
     moduleRegistry.registerExportedModule(new ScopedFileSystemModule(scopedContext));
+
+    // Add SpongyCastle integration
+    try {
+      // If this doesn't throw an exception, we can instantiate the binding.
+      Class.forName("expo.modules.securestore.SecureStoreModule");
+      moduleRegistry.registerExportedModule(new SecureStoreModuleBinding(scopedContext));
+    } catch (ClassNotFoundException e) {
+      // do nothing, if there's no SecureStoreModule we don't need to override it
+    }
 
     // Adding other modules (not universal) to module registry as consumers.
     // It allows these modules to refer to universal modules.

--- a/android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ExpoModuleRegistryAdapter.java
+++ b/android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ExpoModuleRegistryAdapter.java
@@ -4,18 +4,17 @@ import com.facebook.react.bridge.NativeModule;
 import com.facebook.react.bridge.ReactApplicationContext;
 
 import org.json.JSONObject;
-
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Map;
-
 import org.unimodules.adapters.react.ModuleRegistryAdapter;
 import org.unimodules.adapters.react.ModuleRegistryReadyNotifier;
 import org.unimodules.adapters.react.NativeModulesProxy;
 import org.unimodules.adapters.react.ReactModuleRegistryProvider;
 import org.unimodules.core.ModuleRegistry;
-import org.unimodules.core.interfaces.RegistryLifecycleListener;
 import org.unimodules.core.interfaces.InternalModule;
+import org.unimodules.core.interfaces.RegistryLifecycleListener;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
 
 import host.exp.exponent.ExponentManifest;
 import host.exp.exponent.kernel.ExperienceId;
@@ -59,6 +58,7 @@ public class ExpoModuleRegistryAdapter extends ModuleRegistryAdapter implements 
     // Overriding expo-file-system FileSystemModule
     moduleRegistry.registerExportedModule(new ScopedFileSystemModule(scopedContext));
 
+    // Add SpongyCastle integration
     moduleRegistry.registerExportedModule(new SecureStoreModuleBinding(scopedContext));
 
     // ReactAdapterPackage requires ReactContext

